lib/features/map_view/widgets/markers_layer.dart

```diff
@@ -0,0 +1,33 @@
+import "package:flutter/material.dart";
+import "package:flutter_map/flutter_map.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+
+import "../../../utils/where_non_null_iterable.dart";
+import "../controllers/controllers_set.dart";
+import "map_config.dart";
+
+class MarkersConsumerLayer<T extends GoogleNavigable> extends ConsumerWidget {
+  const MarkersConsumerLayer({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final asyncItems = ref
+        .watch(context.mapSourceRepository<T>())
+        .valueOrNull
+        .whereNonNull
```

Comment:
Looks like a good place to put some custom extension ;) Just suggestion

Reply:
this requires both `ref` and `context`, so imo won't be more readable